README.md

```diff
@@ -12,27 +12,93 @@ As mentioned in our [FAQ](https://www.infracost.io/docs/faq), no cloud credentia
+ 
+- infracost-atlantis:atlantis0.16-infracost0.9 with Atlantis v0.18.1 and Infracost v0.9.17 
+- infracost-atlantis:atlantis0.17-infracost0.9 with Atlantis v0.17.6 and Infracost v0.9.17
+- infracost-atlantis:atlantis0.18-infracost0.9 with Atlantis v0.16.1 and Infracost v0.9.17
+- infracost-atlantis:latest with Atlantis v0.18.1 and Infracost v0.9.17
+
+To generate cost estimates, update your Atlantis configuration to add a [custom command](https://www.runatlantis.io/docs/custom-workflows.html#running-custom-commands) that runs Infracost with the required environment variables, such as `INFRACOST_API_KEY`.  The following simple example adds the Infracost cost estimate to the Atlantis output.  See [the examples section](examples) for more advanced configurations.
+
+```
+docker run -p 4141:4141 -e INFRACOST_API_KEY=$INFRACOST_API_KEY \
+  infracost/infracost-atlantis:latest server \
+  --gh-user=MY_GITHUB_USERNAME \
+  --gh-token=MY_GITHUB_TOKEN \
+  --gh-webhook-secret=MY_GITHUB_WEBHOOK_SECRET \
+  --repo-allowlist='github.com/myorg/*' \
+  --repo-config-json='
+    {
+      "repos": [
+        {
+          "id": "/.*/",
+          "workflow": "terraform-infracost"
+        }
+      ],
+      "workflows": {
+        "terraform-infracost": {
+          "plan": {
+            "steps": [
+              "init",
+              "plan",
+              {
+                "run": "terraform show -json $PLANFILE > $SHOWFILE"
+              },
+              {
+                "run": "echo \"#####\" && echo && echo Infracost output:"
+              },
+              {
+                "run": "infracost diff --path $SHOWFILE --no-color --log-level=warn"   
+              }
+            ]
+          }
+        }
+      }
+    }
+  '
+```
+
+To test, send a new pull request to change something in Terraform that costs money; a comment should be posted on the pull request by Atlantis.  Expand the Show Output section, at the bottom of which you should see the Infracost output.
+
+## Option 2: Install in Docker
+
+Instead of baking Infracost into the Dockerfile, it can be installed dynamically in a pre-workflow hook such as:
 
-1. [This Docker image](https://hub.docker.com/repository/docker/infracost/infracost-atlantis/) extends the Atlantis image by adding the Infracost CLI and the [`infracost_atlantis_diff.sh`](https://github.com/infracost/infracost/blob/master/scripts/ci/atlantis_diff.sh) script. If you already use a custom Docker image for Atlantis, copy the `RUN` commands from [this Dockerfile](https://github.com/infracost/infracost-atlantis/blob/master/Dockerfile) into your Dockerfile.
+```yaml
+repos:
+  - id: /.*/
+    workflow: terraform-infracost
+    pre_workflow_hooks:
+      # Install Infracost
+      - run: |
+          /tmp/infracost --version || curl -L https://github.com/infracost/infracost/releases/latest/download/infracost-linux-amd64.tar.gz --output infracost.tar.gz && \
```

Comment:
I couldn't think of a great way to make it work with `infracost`.  The best I came up with is that we could put the binary in `/home/atlantis/bin`, then add an `env` step to the workflow steps that sets `PATH=$PATH:/home/atlantis/bin`.  If that seems better or if you have another idea I'm open to it.

Reply:
> Also, are we worried changing the exit code here will cause issues for existing users?

I'm not following what you mean

Reply:
> Also, are we worried changing the exit code here will cause issues for existing users?

Sorry. I think my assumption was wrong. I was assuming the `/tmp/infracost --version ||` was to check if the latest version of infracost is being used and it would:
 * Return 0 (success) if infracost is installed and up-to-date
 * Return 1 (fail) if infracost is installed but not up-to-date
 
 So I was worried that returning an error on `infracost --version` for out-of-date versions would mess with any existing user scripts if they already run `infracost --version`.

Reply:
> I couldn't think of a great way to make it work with `infracost`. 

Yeah, I think the `tmp` approach makes sense since there's no perm issues and no need for extra steps. My comment was unclear, and I was meaning if we need to update the `infracost` code's `--version` flag as above.

Reply:
I see, yeah I wasn't thinking about the need to update to the latest version.  What do you think about looking for the "A new version is available" string?:

```bash
/tmp/infracost --version && [ $(/tmp/infracost --version 2>&1 | grep -c "A new version of Infracost is available") = 0 ] || <downalod and install>
```

Like you suggest, I don't think we want to make `infracost --version` start failing, but maybe we could add a `--version-check` flag?  OPA has a separate `version` command so you can do something like `opa version --check`.

Reply:
> What do you think about looking for the "A new version is available" string?

👍 

> you can do something like `opa version --check`

This is style!